lms/djangoapps/certificates/models.py

```diff
@@ -574,3 +579,92 @@ def get_config(cls):
         instance = cls.current()
         json_data = json.loads(instance.configuration) if instance.enabled else {}
         return json_data
+
+
+class BadgeAssertion(models.Model):
+    """
+    Tracks badges on our side of the badge baking transaction
+    """
+    user = models.ForeignKey(User)
+    course_id = CourseKeyField(max_length=255, blank=True, default=None)
+    data = JSONField()
+
+    @property
+    def image_url(self):
+        """
+        Get the image for this assertion.
+        """
+        return self.data['json']['image']
+
+    class Meta(object):
+        """
+        Meta information for Django's construction of the model.
+        """
+        unique_together = (('course_id', 'user'),)
+
+
+class BadgeImageConfiguration(models.Model):
+    """
+    Contains the configuration for badges for a specific mode. The mode
+    """
+    mode = models.CharField(
+        max_length=125,
+        help_text=_(u"The course mode this image is for, like 'verified' or 'honor'."),
+        unique=True,
+    )
+    icon = models.ImageField(
+        # Actual max is 256KB, but need overhead for badge baking. This should be more than enough.
+        help_text=_(
+            u"Badge images must be square PNG files. To use with Mozilla Backpack, the filesize should be "
+            u"under 250 kilobytes."
+        ),
+        upload_to='badges',
+    )
+    default = models.BooleanField(
+        help_text=_(
+            u"Set this if this image should be the default image for any modes not specified. You may have only one "
+            u"default image."
+        )
+    )
+
```

Comment:
Can you add a data migration to include default badges? While we wait for UI to send the final badges, it can be mock images, this way we would just have to swap them when they are ready.